livre/lib/data/BoundingAxis.cpp

```diff
@@ -0,0 +1,368 @@
+/* Copyright (c) 2016, EPFL/Blue Brain Project
+ *                     Grigori Chevtchenko <grigori.chevtchenko@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <livre/lib/data/BoundingAxis.h>
+
+#include <livre/core/data/VolumeInformation.h>
+
+#include <eq/gl.h>
+
+namespace livre
+{
+
+enum LineType
+{
+    BBOX = 0,
+    AXIS
+};
+
+BoundingAxis::BoundingAxis( const VolumeInformation& volInfo )
+  : _nVertices( 0 )
+{
+    BoundingAxisData bbAxisData;
+    bbAxisData.volInfo = volInfo;
+    const Vector4f color( 0.5, 1.0, 0.69, 1.0 );
+
+    const float x = volInfo.worldSize.x() * 0.5f;
+    const float y = volInfo.worldSize.y() * 0.5f;
+    const float z = volInfo.worldSize.z() * 0.5f;
+
+    const float axisSizes[ 3 ] = { x, y, z };
+    std::vector< Vector3f > vertices;
+    std::vector< Vector3f > normals;
+
+    for( int32_t i = -1; i < 2; i += 2 )
+    {
+        for( int32_t j = 0; j < 3; ++j )
+        {
+            for( int32_t k = 0; k < 4; ++k )
+            {
+                int32_t sign1 = (int32_t)( k / 2 ) * 2 - 1;
+                int32_t sign2 = ((int32_t)(( k + 1 ) / 2 ) % 2 ) * 2 - 1;
+
+                Vector3f vertex;
+                Vector3f normal;
+                int32_t axis = j;
+                vertex[ axis ] = -i * axisSizes[ axis ];
+                normal[ axis ] = i;
+
+                axis = (j + 1) % 3;
+                vertex[ axis ] = sign1 * axisSizes[ axis ];
+                normal[ axis ] = 0.0;
+
+                axis = (j + 2) % 3;
+                vertex[ axis ] = sign2 * axisSizes[ axis ];
+                normal[ axis ] = 0.0;
+
+                vertices.push_back( vertex );
+                normals.push_back( normal );
+
+                if( k != 0 )
+                {
+                    vertices.push_back( vertex );
+                    normals.push_back( normal );
+                }
+
+                if( k == 3 )
+                {
+                    vertices.push_back( vertices[ vertices.size() - 7 ]);
+                    normals.push_back( normals[ normals.size() - 7 ]);
+                }
+            }
+        }
+    }
+
+    // There is a possible optimization here, it is possible to append directly
+    // in the code lines [ 50 - 89 ].
```

Comment:
This technique will not work for bbAxisData.normals, bbAxisData.normals2, bbAxisData.color, bbAxisData.types. There is a way to do it for everything in the loop but it will take more time and the code will be even more complicated. Here you are only copying 48 vertexes...

I just put a comment for future work.